llvm/lib/Transforms/Utils/LoopUtils.cpp

```diff
@@ -415,6 +417,87 @@ MDNode *llvm::updateIterCounts(LLVMContext &Context, MDNode *LoopID,
   return NewLoopID;
 }
 
+const MDNode *llvm::getLoopID(const BasicBlock *LoopBB) {
+  if (!LoopBB)
+    return nullptr;
+  const Instruction *TI = LoopBB->getTerminator();
+  if (!TI)
+    return nullptr;
+
+  const MDNode *LoopID = TI->getMetadata(LLVMContext::MD_loop);
+  return LoopID;
+}
```

Comment:
This seems to be a new function. Is that different from `LoopT *LoopInfoBase::getLoopFor(const BlockT *BB)`? I think we should avoid adding such low-level helpers if there's already a similar one.

Reply:
I left this function alone in `AIELoopUtils.cpp`